### PR TITLE
Move spdlog to main 

### LINKED ISSUE #111297: Sandbox: provide a functional logger that does not depend on spdlog in renderer
Today we require `spdlog` from the renderer as native module. It is pretty clear that once we enable `sandbox`, this dependency cannot be met anymore. This issue is to discuss alternatives, some thoughts:
* use shared process for logging (ideal because it would not block the main process)
* just use a file based logger that uses file service (I guess that goes against why `spdlog` was introduced at first)

//cc @deepak1556 @joaomoreno

### PR DESCRIPTION
**Pull request by sandy081**
This PR fixes https://github.com/microsoft/vscode/issues/111297

- Register `ILoggerService` on main side. This can provide spdlog logger given a file.
- Create Logger channel to get logger from main
- In renderer, register Logger service that uses above logger channel
- Use above logger service to create logger for logging in renderer
- Use above logger service to create logger for output channels

### CODE REVIEW

**Comment by sandy081** [comment]
Just FYI - because of refactoring (rename), it also needs corresponding adoptions in the server code.

File: src/vs/workbench/workbench.desktop.main.ts [hunk 1]
```diff
@@ -111,7 +111,7 @@ registerSingleton(ISharedProcessService, SharedProcessService, true);
 //#region --- workbench contributions
 
 // Output
-import 'vs/workbench/contrib/output/electron-browser/outputChannelModelService';
+import 'vs/workbench/contrib/output/electron-sandbox/outputChannelModelService';
```
Comment (bpasero) [inline src/vs/workbench/workbench.desktop.main.ts:114]: Please move this import over to `src/vs/workbench/workbench.sandbox.main.ts `

**Review by bpasero — APPROVED** [review]
I did not do a deep review, mainly focused on changes around where the log service is setup (main) and used (renderer). Only found one nit for the import that can now be moved to `sandbox`.

I trust you for the changes in the various logger related files 👍 

Thanks for moving this forward 🚀

**Review by bpasero — CHANGES_REQUESTED** [review]
Oh, have to reject, there are layer issues:

```
[valid-layers-check] [build/lib/layersChecker.ts]: Reference to 'Buffer' from '@types/node' violates layer '**/vs/**/electron-sandbox/**' (/agent/_work/1/s/src/vs/workbench/contrib/output/electron-sandbox/outputChannelModelService.ts (65,37)
[valid-layers-check] [build/lib/layersChecker.ts]: Reference to 'from' from '@types/node' violates layer '**/vs/**/electron-sandbox/**' (/agent/_work/1/s/src/vs/workbench/contrib/output/electron-sandbox/outputChannelModelService.ts (65,44)
[valid-layers-check] [build/lib/layersChecker.ts]: Reference to 'Buffer' from '@types/node' violates layer '**/vs/**/electron-sandbox/**' (/agent/_work/1/s/src/vs/workbench/contrib/output/electron-sandbox/outputChannelModelService.ts (92,47)
[valid-layers-check] [build/lib/layersChecker.ts]: Reference to 'from' from '@types/node' violates layer '**/vs/**/electron-sandbox/**' (/agent/_work/1/s/src/vs/workbench/contrib/output/electron-sandbox/outputChannelModelService.ts (92,54)
```

**Comment by sandy081** [comment]
Thanks will do the necessary change after merging.

**Comment by sandy081** [comment]
Fixed layering issue and also moved to sandbox.main

**Review by bpasero — APPROVED** [review]

**Pull request merged**